Cleanup multicluster doc

frankbu opened the pull request:
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

vadimeisenbergibm commented:
@frankbu I do not like the concept of "Multiple control planes" since "Shared control planes" can also be multiple. Like here:

![PrimarySecondaryMulticluster_4clusters-4](https://user-images.githubusercontent.com/16784982/60754037-0cf12180-9fe4-11e9-81ca-f329ff1d8a59.png)

Once we decided that we call one pattern "Shared", I like it, let's call the other patten the opposite of "Shared": _Nonshared_ or one of the antonyms https://www.merriam-webster.com/thesaurus/shared.

vadimeisenbergibm commented on content/docs/setup/kubernetes/additional-setup/config-profiles/index.md:
```diff
@@ -71,7 +71,7 @@ Istio provides two additional built-in configuration profiles that are used excl
 
 1. **remote**: used for configuring remote clusters of a
-    multicluster mesh with a [single control plane topology](/docs/concepts/multicluster-deployments/#single-control-plane-topology).
+    multicluster mesh with a [single control plane topology](/docs/concepts/multicluster-deployments/#shared-control-plane-topology).
```
[single control plane topology] -> [shared control plane topology]

vadimeisenbergibm commented on content/docs/setup/kubernetes/install/multicluster/shared-gateways/index.md:
```diff
@@ -1,19 +1,22 @@
-and using Istio's _Split-horizon EDS (Endpoints Discovery Service)_ feature (introduced in Istio 1.1) to
+Follow this guide to configure a multicluster mesh with a
+[shared control plane topology](/docs/concepts/multicluster-deployments/#shared-control-plane-topology)
+and using Istio's _Split-horizon EDS (Endpoints Discovery Service)_ feature to
```
Split horizon EDS is an implementation detail, I am not sure we should mention it at all. I do not think it will help the users in any way. We can mention it in the concept page, but I do not think that it is relevant in a task the users exercise.

vadimeisenbergibm commented on content/docs/setup/kubernetes/install/multicluster/shared-gateways/index.md:
```diff
@@ -1,19 +1,22 @@
+and using Istio's _Split-horizon EDS (Endpoints Discovery Service)_ feature to
 route service requests to other clusters via their ingress gateway.
-Split-horizon EDS enables Istio to route requests to different endpoints, depending on the location of the request source.
+Split-horizon EDS enables Istio to route requests to different endpoints,
```
The same comment as https://github.com/istio/istio.io/pull/4638/files#r300825440

vadimeisenbergibm commented:
@frankbu Great work, thank you for this endeavor! I left several comments.

frankbu commented:
@vadimeisenbergibm thanks for the review. I've addressed all your comments, except the "multiple control planes" one, which I'd prefer to leave as is, at least for now. All the other changes are aligned with more-or-less agreed terminology and are mostly about reorging the material. We can rename "multiple control planes" in a future PR, after we have agreement on a better name.

vadimeisenbergibm commented:
/lgtm

sdake commented on content/docs/concepts/multicluster-deployments/index.md:
```diff
@@ -100,16 +102,16 @@ same IP address.
+[single-network shared control plane instructions](/docs/setup/kubernetes/install/multicluster/shared-vpn/).
 
-#### Single control plane without VPN connectivity
+#### Multi-network shared control plane
```
for a followup - the key thing to explain about multi-network is that the networks are isolated, not that a gateway is necessary.  A gateway is always necessary if the networks are isolated.  I am not sure we want to use the terminology "Isolated" though - it has negative connotations for the title.  For explanation text, it would be very valuable.

sdake commented on content/docs/examples/multicluster/icp/index.md:
```diff
@@ -4,13 +4,13 @@ description: Example multicluster mesh over two IBM Cloud Private clusters.
 [IBM Cloud Private](https://www.ibm.com/cloud/private) clusters
 and then compose them into a multicluster mesh using a
-[single control plane with VPN connectivity](/docs/concepts/multicluster-deployments/#single-control-plane-with-vpn-connectivity)
+[single-network shared control plane](/docs/concepts/multicluster-deployments/#single-network-shared-control-plane)
```
as mentioned prior, single-network needs refinement imo.  This could work as a label, however, the context of the label should be set somewhere in the documentation more completely.

frankbu replied:
Maybe single-network and multi-network should be glossary entries that we reference.
^^^ @adammil2000

sdake commented on content/docs/examples/multicluster/icp/index.md:
```diff
@@ -148,7 +148,7 @@ across all nodes in the two IBM Cloud Private Clusters.
 ## Install Istio for multicluster
 
-[Follow the VPN-based multicluster installation steps](/docs/setup/kubernetes/install/multicluster/vpn/) to install and configure
+[Follow the VPN-based multicluster installation steps](/docs/setup/kubernetes/install/multicluster/shared-vpn/) to install and configure
```
Follow the "single-network shared control plane installation steps"?  See there is a bit of inconsistency.

sdake commented on content/docs/setup/kubernetes/install/multicluster/gateways/index.md:
```diff
@@ -1,24 +1,28 @@
-when services in a Kubernetes cluster use gateways to communicate with services in other clusters.
+Follow this guide to install an Istio
+[multicluster service mesh](/docs/concepts/multicluster-deployments/#multicluster-service-mesh)
+with locally deployed Istio control planes in every cluster and using gateways to
```
locally/individually

sdake replied:
@frankbu re the wording changes, I think as you suggested we should leave the release notes as is.  Can you address this non-release notes change?  The wording of individually is a little misleading.

sdake commented on content/docs/setup/kubernetes/install/multicluster/shared-vpn/index.md:
```diff
@@ -1,13 +1,14 @@
-title: VPN Connectivity
-description: Install an Istio mesh across multiple Kubernetes clusters with direct network access to remote pods.
+title: Shared control plane (single-network)
+description: Install an Istio mesh across multiple Kubernetes clusters with a shared control plane and VPN connectivity between clusters.
```
its not just VPN connectivity - any routed pod network will do.  This could be a VPN, or a VPC, or even plain old l3 routing in an enterprise.

rvennam commented on content/docs/concepts/multicluster-deployments/index.md:
```diff
@@ -42,9 +42,11 @@ Istio control plane. There are two possible deployment approaches:
 
 1. Multiple synchronized Istio control planes that have replicated service and routing configurations.
```
This wording implies that the control planes are synching and share config themselves. However, they are independent (not necessarily identical) control planes and operators can pick and choose what services they want to connect using gateways and service entries. There is nothing currently synchronized automatically. Compared to the shared control plane approach, there is less manual replication.
